### PR TITLE
[HttpClient] Add `HttpOptions::addHeader` method

### LINKED ISSUE #19236: [HttpClient] Add HttpOptions->addHeader as a shortcut to add an header …
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#52946
| PR author(s) | @Dean151
| Merged in    | 7.1

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.1 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by Dean151**
Fixes #19236

Not sure of the documentation format. Open to suggestion to improve it :)

### CODE REVIEW

File: http_client.rst [hunk 1]
```diff
@@ -154,6 +154,15 @@ brings most of the available options with type-hinted getters and setters::
             ->toArray()
     );
 
+.. tip::
+
+    Be aware that calling :method:`Symfony\\Component\\HttpClient\\HttpOptions::setHeaders` overrides all previous headers.
```
Comment (xabbuh) [inline http_client.rst]: I would update the code example above to call both the `setHeaders()` and `setHeader()` method and add inline comments explaining the difference.

Comment (Dean151) [reply http_client.rst]: I've made a change in that direction. Tell me if there is anything I can improve further ;)

File: http_client.rst [hunk 2]
```diff
@@ -150,10 +150,17 @@ brings most of the available options with type-hinted getters and setters::
     $this->client = $client->withOptions(
         (new HttpOptions())
             ->setBaseUri('https://...')
+            // Set headers replace *all* headers at once, and deletes the headers you do not provide
```
Comment (xabbuh) [inline http_client.rst]:
```suggestion
            // setHeaders() replaces *all* headers at once, and deletes the headers you do not provide
```

File: http_client.rst [hunk 3]
```diff
@@ -150,10 +150,17 @@ brings most of the available options with type-hinted getters and setters::
...
             ->setHeaders(['header-name' => 'header-value'])
+            // Alternatively, add or replace only one header to an existing HttpOptions object.
```
Comment (xabbuh) [inline http_client.rst]:
```suggestion
            // add or replace a single header using addHeader()
```

**Comment by OskarStark** [comment]
Thank you Thomas.

**Pull request merged**